### ISSUE TITLE
WPF GridView Focus issue

### ISSUE THREAD

**Issue by TomQv** [picoe/Eto#831]
I have a form with a TreeGridView and a GridView.
None of them has the focus currently.

1. I click on an item in the TreeGridView. It gets selected and TreeGridView.GotFocus is fired
2. Now I click on an item in the GridView. It gets selected and GridView.GotFocus is fired.
3. Now I click again on the item in TreeGridView which I selected in 1. TreeGridView.GotFocus is Not fired.
4. Now I click again on the item in GridView which I selected in 2. GridView.GotFocus is not fired, but  TreeGridView.GotFocus, which was supposed to fire on 3. is now fired.
5. repeating 3. and 4. now always fires GotFocus for the wrong grid.

This is with WPF. 
WinForms is fine.
Didn't try Mac yet.

Btw. how to move focus between grid-controls with the Tab-Key? 
Is it OS-specific, that tab only moves to the next cell?

**Comment by cwensley** [comment]
Hey @TomQv, thanks for the report.

Yes, the Tab key behaviour is platform specific.

**Comment by TomQv** [comment]
ok for the tab-key, but 1.-5. is most likely a bug.

**Comment by cwensley** [comment]
Yes I agree!  Sorry, I didn't mention that. (;

**Comment by TomQv** [comment]
I think, the problem is in Eto.Wpf.Forms.WpfFrameworkElement.AttachEvent(string id)-

Changing GotFocusEvent and LostFocusEvent like this and it works fine:

```
case Eto.Forms.Control.GotFocusEvent:
	Control.GotFocus += (sender, e) =>
	{
	   Callback.OnGotFocus(Widget, EventArgs.Empty);
	};
	break;

case Eto.Forms.Control.LostFocusEvent:

	Control.LostFocus += (sender, e) =>
	{
		Callback.OnLostFocus(Widget, EventArgs.Empty);
	};
	break;

```

**Comment by cwensley** [comment]
We can't use GotFocus/LostFocus since it is not equivalent to keyboard focus on other platforms.  See https://stackoverflow.com/questions/1320816/keyboard-focus-vs-logical-focus-in-wpf.

**Comment by cwensley** [comment]
Yes, so we should fix `HasFocus` instead of working around its bug.
